src/include/mpir_contextid.h

```diff
@@ -8,9 +8,19 @@
 #ifndef MPIR_CONTEXTID_H_INCLUDED
 #define MPIR_CONTEXTID_H_INCLUDED
 
+#ifdef HAVE_EXTENDED_CONTEXT_BITS
+#define MPIR_CONTEXT_ID_T_DATATYPE MPI_UINT32_T
+typedef uint32_t MPIR_Context_id_t;
+#define MPIR_INVALID_CONTEXT_ID ((MPIR_Context_id_t)0xffffffff)
+#define MPIR_CONTEXT_ID_BITS 20
+#define CONTEXT_ID_FMT PRIu32
+#else
 #define MPIR_CONTEXT_ID_T_DATATYPE MPI_UINT16_T
 typedef uint16_t MPIR_Context_id_t;
 #define MPIR_INVALID_CONTEXT_ID ((MPIR_Context_id_t)0xffff)
+#define MPIR_CONTEXT_ID_BITS (sizeof(MPIR_Context_id_t)*8)      /* 8 --> CHAR_BITS eventually */
```

Comment:
How about define it directly to 16, as the 20-bit case?

Reply:
Sure, probably better to be explicit. Will change.